python/lsst/ts/logging_and_reporting/web_app/services/scheduler_service.py

```diff
@@ -0,0 +1,768 @@
+        observatory = ModelObservatory(nside=nside, init_load_length=1)
+        observatory.sky_model.load_length = 1
+
+    unique_nights = sorted(visits["day_obs"].unique())
+
+    conditions_list = []
+    for day_obs in unique_nights:
+        night_date = datetime.strptime(str(day_obs), '%Y%m%d').date()
+        night_events = schedview.compute.astro.night_events(night_date=night_date, site=observatory.location,
+                                                            timezone=timezone)
+        # start_time = Time(night_events.loc["sunset","UTC"])
+        end_time = Time(night_events.loc["sunrise","UTC"])
+        observatory.mjd = end_time.mjd
+        conditions_list.append(observatory.return_conditions())
+
+    # Footprint outline
+    footprint_regions = get_current_footprint(nside)[1]
+    footprint_regions[np.isin(footprint_regions, ["bulgy","lowdust"])] = "WFD"
+    footprint_regions[np.isin(footprint_regions,
+                              ["LMC_SMC","dusty_plane","euclid_overlap","nes","scp","virgo"])
+                              ] = "other"
+    footprint_outline = find_healpix_area_polygons(footprint_regions)
+    tiny_loops = footprint_outline.groupby(["region","loop"]).count().query("RA<10").index
+    footprint_outline = footprint_outline.drop(tiny_loops)
+
+    data = {
+        "visits": visits,
+        "footprint": None,
+        "footprint_outline": footprint_outline,
+        "conditions_list": conditions_list,
+        "applet_mode": applet_mode,
+        "theme": theme,
+    }
+
+    # Call plotting function
+    if applet_mode or planisphere_only:
+        vmap = plot_visit_skymaps(map_classes=[Planisphere], **data)
+    else:
+        vmap = plot_visit_skymaps(map_classes=[ArmillarySphere, Planisphere], **data)
+
+    return vmap, data
+
+
+def my_map_visits_over_hpix(
+    visits,
+    conditions,
+    map_hpix,
+    plot=None,
+    scale_limits=None,
+    palette=colorcet.blues,
+    map_class=Planisphere,
+    prerender_hpix=True,
+):
+    """Plot visit locations over a healpix map.
+
+    Parameters
+    ----------
+    visits : `pd.DataFrame`
+        The table of visits to plot, with columns matching the opsim database
+        definitions.
+    conditions : `rubin_scheduler.scheduler.features.Conditions`
+        An instance of a rubin_scheduler conditions object.
+    map_hpix : `numpy.array`
+        An array of healpix values
+    plot : `bokeh.models.plots.Plot` or `None`
+        The bokeh plot on which to make the plot. None creates a new plot.
+        None by default.
+    scale_limits : `list` of `float` or `None`
+        The scale limits for the healpix values. If None, use zscale to set
+        the scale.
+    palette : `str`
+        The bokeh palette to use for the healpix map.
+    map_class : `class`, optional
+        The class of map to use.  Defaults to uranography.Planisphere.
+    prerender_hpix : `bool`, optional
+        Pre-render the healpix map? Defaults to True
```

Comment:
```suggestion
    visits : `pd.DataFrame`
        The table of visits to plot, with columns matching the opsim database
        definitions.
    conditions : `rubin_scheduler.scheduler.features.Conditions`
        An instance of a rubin_scheduler conditions object.
    map_hpix : `numpy.array`
        An array of healpix values.
    plot : `bokeh.models.plots.Plot` or `None`
        The bokeh plot on which to make the plot. None creates a new plot.
        None by default.
    scale_limits : `list` [`float`] or `None`
        The scale limits for the healpix values. If None, use zscale to set
        the scale.
    palette : `str`
        The bokeh palette to use for the healpix map.
    map_class : `class`, optional
        The class of map to use.  Defaults to uranography.Planisphere.
    prerender_hpix : `bool`, optional
        Pre-render the healpix map? Defaults to True
```
https://developer.lsst.io/python/numpydoc.html#sequence-types